--- a/applications/luci-upnp/luasrc/model/cbi/upnp/upnp.lua
--- b/applications/luci-upnp/luasrc/model/cbi/upnp/upnp.lua
@@ -40,14 +40,22 @@
 	end
 end
 
-s:taboption("general", Flag, "enable_upnp", translate("Enable UPnP functionality")).default = "1"
-s:taboption("general", Flag, "enable_natpmp", translate("Enable NAT-PMP functionality")).default = "1"
+ni = s:taboption("general", Flag, "enable_upnp", translate("Enable UPnP functionality"))
+ni.default = "1"
+ni.rmempty = false
 
-s:taboption("general", Flag, "secure_mode", translate("Enable secure mode"),
-	translate("Allow adding forwards only to requesting ip addresses")).default = "1"
+ni = s:taboption("general", Flag, "enable_natpmp", translate("Enable NAT-PMP functionality"))
+ni.default = "1"
+ni.rmempty = false
 
-s:taboption("general", Flag, "log_output", translate("Enable additional logging"),
+ni = s:taboption("general", Flag, "secure_mode", translate("Enable secure mode"),
+	translate("Allow adding forwards only to requesting ip addresses"))
+ni.default = "1"
+ni.rmempty = false
+
+ni = s:taboption("general", Flag, "log_output", translate("Enable additional logging"),
 	translate("Puts extra debugging information into the system log"))
+ni.rmempty = false
 
 s:taboption("general", Value, "download", translate("Downlink"),
 	translate("Value in KByte/s, informational only")).rmempty = true
--- a/applications/luci-upnp/root/etc/uci-defaults/luci-upnp
+++ a/applications/luci-upnp/root/etc/uci-defaults/luci-upnp
@@ -1,9 +1,6 @@
 #!/bin/sh
 
-/etc/init.d/miniupnpd enabled && {
-	/etc/init.d/miniupnpd stop
-	/etc/init.d/miniupnpd disable
-}
+/etc/init.d/miniupnpd enabled
 
 rm -f /tmp/luci-indexcache
 exit 0
